source/dofs/dof_tools.cc

```diff
@@ -1093,6 +1140,9 @@ namespace DoFTools
            ExcDimensionMismatch(subdomain_association.size(),
                                 dof_handler.n_dofs()));
 
+    Assert(dof_handler.n_dofs() > 0,
+           ExcMessage("Number of DoF is not positive. "
+                      "This could happen when the function is called before NumberCache is written."));
```

Comment:
that's a good question. I guess a single cell will be assigned to one of the processors (parmetis) and all of its dofs will be locally owned. The assert shall not be violated though, as the total number of dofs still have to be more than zero.